### PR TITLE
Present `WpApiError.errorMessage` instead of `localizedDescription`

### PR DESCRIPTION
**Pull request by crazytonyli**
What says in the title.

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/Extensions/WpApiError+Localized.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import Foundation
+import WordPressAPI
+
+/// `WpApiError` confirms to `LocalizedError`, but with an implementation that's not suitable for displaying on UI.
+/// When presenting `WpApiError`, we should use the `errorMessage` function instead of `localizedDescription`.
```
Comment (crazytonyli) [inline WordPress/Classes/Extensions/WpApiError+Localized.swift:5]: We'll be able to implement `LocalizedError` properly in wordpress-rs, once/if https://github.com/mozilla/uniffi-rs/pull/2319 is merged and released.

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/wordpress-mobile/WordPress-iOS/milestone/303">25.6</a>. The due date for this milestone has already passed.<br />
Please assign it to a milestone with a later deadline or check whether the release for this milestone has already been finished.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WordPress/Classes/Extensions/WpApiError+Localized.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import Foundation
+import WordPressAPI
+
+/// `WpApiError` confirms to `LocalizedError`, but with an implementation that's not suitable for displaying on UI.
```
Comment (kean) [inline WordPress/Classes/Extensions/WpApiError+Localized.swift]:
```suggestion
/// `WpApiError` conforms to `LocalizedError`, but with an implementation that's not suitable for displaying on UI.
```

File: WordPress/Classes/Extensions/WpApiError+Localized.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+/// When presenting `WpApiError`, we should use the `errorMessage` function instead of `localizedDescription`.
+extension WpApiError {
+    var errorMessage: String {
+        switch self {
+        case .InvalidHttpStatusCode, .SiteUrlParsingError, .UnknownError:
+            return NSLocalizedString("Something went wrong", comment: "A generic error message")
```
Comment (kean) [inline WordPress/Classes/Extensions/WpApiError+Localized.swift]:
```suggestion
            return SharedStrings.Error.generic
```

**Review by kean — APPROVED** [review]
Since there is no way of telling when https://github.com/mozilla/uniffi-rs/pull/2319/files lands, this looks OK.

Perhaps there is a way to re-wrap the auto-generated `WpApiError` into something else that's more useful for the consumer? The app has generalized code that relies on `localizedDescription` for errors:

```swift
extension EmptyStateView where Label == SwiftUI.Label<Text, Image>, Description == Text?, Actions == Button<Text> {
    static func failure(error: Error, onRetry: @escaping () -> Void) -> Self {
        EmptyStateView {
            Label(SharedStrings.Error.generic, systemImage: "exclamationmark.circle")
        } description: {
            Text(error.localizedDescription)
        } actions: {
            Button(SharedStrings.Button.retry, action: onRetry)
        }
    }
}
```

On a side note, the total disregard for Swift API Naming Conventions hurts 😭

**Comment by crazytonyli** [comment]
> Perhaps there is a way to re-wrap the auto-generated WpApiError into something else that's more useful for the consumer? 

Can you elaborate on this? What functions would the wrapper have?

> On a side note, the total disregard for Swift API Naming Conventions hurts 😭

Agreed. I can probably look into this.

**Comment by kean** [comment]
> Can you elaborate on this? What functions would the wrapper have?

Basically just proper conformance to `LocalizedError` so it could be used with `EmptyStateView` and other places that rely on it.

**Comment by crazytonyli** [comment]
Ah of course. The uniffi-rs PR has been merged. So we should not be too far from implementing `LocalizedError` directly on `WpApiError`.

**Comment by crazytonyli** [comment]
⬆️ Rebase to resolve conflicts.

**Pull request merged**